mlir/lib/Mitigation/Transforms/MitigationMethods/Zne.cpp

```diff
@@ -0,0 +1,310 @@
+                    auto adjointOp = builder.create<quantum::AdjointOp>(loc, qregType,
+                                                                        fnWithoutMeasurementsQreg);
+                    Region *adjointRegion = &adjointOp.getRegion();
+                    Block *adjointBlock = builder.createBlock(adjointRegion, {}, qregType, loc);
+
+                    std::vector<Value> argsAndQregAdjoint = {fnFoldedOp.getArguments().begin(),
+                                                             fnFoldedOp.getArguments().end()};
+                    argsAndQregAdjoint.pop_back();
+                    argsAndQregAdjoint.push_back(adjointBlock->getArgument(0));
+                    Value fnWithoutMeasurementsAdjointQreg =
+                        builder
+                            .create<func::CallOp>(loc, fnWithoutMeasurementsOp, argsAndQregAdjoint)
+                            .getResult(0);
+                    builder.create<quantum::YieldOp>(loc, fnWithoutMeasurementsAdjointQreg);
+                    builder.setInsertionPointAfter(adjointOp);
+                    builder.create<scf::YieldOp>(loc, adjointOp.getResult());
+                })
+            .getResult(0);
+    std::vector<Value> argsAndRegMeasurement = {fnFoldedOp.getArguments().begin(),
+                                                fnFoldedOp.getArguments().end()};
+    argsAndRegMeasurement.pop_back();
+    argsAndRegMeasurement.push_back(loopedQreg);
+    ValueRange funcFolded =
+        rewriter.create<func::CallOp>(loc, fnWithMeasurementsOp, argsAndRegMeasurement)
+            .getResults();
+    rewriter.create<func::ReturnOp>(loc, funcFolded);
+    return SymbolRefAttr::get(ctx, fnFoldedName);
+}
+FlatSymbolRefAttr ZneLowering::getOrInsertQuantumAlloc(Location loc, PatternRewriter &rewriter,
+                                                       mitigation::ZneOp op)
+{
+    // Quantum Alloc function
+    MLIRContext *ctx = rewriter.getContext();
+    OpBuilder::InsertionGuard guard(rewriter);
+    ModuleOp moduleOp = op->getParentOfType<ModuleOp>();
+    Type qregType = quantum::QuregType::get(rewriter.getContext());
+
+    std::string fnAllocName = op.getCallee().str() + ".quantumAlloc";
+    if (moduleOp.lookupSymbol<func::FuncOp>(fnAllocName)) {
+        return SymbolRefAttr::get(ctx, fnAllocName);
+    }
+    Type i64Type = rewriter.getI64Type();
+    FunctionType fnAllocType = FunctionType::get(ctx, /*inputs=*/
+                                                 i64Type,
+                                                 /*outputs=*/qregType);
+    func::FuncOp fnAlloc = rewriter.create<func::FuncOp>(loc, fnAllocName, fnAllocType);
+    fnAlloc.setPrivate();
+    Block *allocBloc = fnAlloc.addEntryBlock();
+    rewriter.setInsertionPointToStart(allocBloc);
+    Value nQubits = allocBloc->getArgument(0);
+    IntegerAttr intAttr;
+    auto qreg = rewriter.create<quantum::AllocOp>(loc, qregType, nQubits, intAttr);
```

Comment:
What's the use case of `intAttr`?

Reply:
The C++ interface does not like the optional nature of the number of qubits, you have to provide all arguments and therefore give an empty one.

Reply:
Okay then, can we update `IntegerAttr intAttr` to `IntegerAttr intAttr{}` to trigger its default constructor. This'll ensure that we're passing an initialized value.